From kay.sievers@vrfy.org  Mon Apr  6 12:14:46 2009
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Fri, 03 Apr 2009 19:04:15 +0200
Subject: driver core: allow non-root users to listen to uevents
To: Greg KH <greg@kroah.com>
Message-ID: <1238778255.16630.4.camel@poy>

From: Kay Sievers <kay.sievers@vrfy.org>

Users can read sysfs files, there is no reason they should not be
allowed to listen to uevents.  This lets xorg and other userspace
programs properly get these messages without having to be root.

Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject_uevent.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/kobject_uevent.c
+++ b/lib/kobject_uevent.c
@@ -328,7 +328,7 @@ static int __init kobject_uevent_init(vo
 		       "kobject_uevent: unable to create netlink socket!\n");
 		return -ENODEV;
 	}
-
+	netlink_set_nonroot(NETLINK_KOBJECT_UEVENT, NL_NONROOT_RECV);
 	return 0;
 }
 
